src/object/srv_obj.c

```diff
@@ -1753,10 +1757,10 @@ obj_ioc_begin(uint32_t rpc_map_ver, uuid_t pool_uuid,
 	if (rc != 0)
 		return rc;
 
-	/** Update sensors */
+	/** increment active request counter and start the chrono */
 	tls = obj_tls_get();
-	d_tm_mark_duration_start(&tls->ot_op_lat[opc], 0, NULL);
 	d_tm_increment_gauge(&tls->ot_op_active[opc], 1, NULL);
+	ioc->ioc_start_time = daos_get_ntime();
```

Comment:
Are there multiple opcodes in flight per ioc?  If so, I am not following how setting the ioc_start_time here gives a better start time for this opcode than before.  If another xs can issue an overlapping opc, won't that overwrite the ioc_start_time stored at the io context level when it calls obj_ioc_begin for the same ioc again?

Reply:
I see that Wangdi says it is OK.  I'm not familiar enough with the structure here, so I am just asking the questions.  If you are both satisfied that you are getting the correct time bounds, that's fine with me.  The use of the gauge is correct.

Reply:
ioc is per request, while tls is per service thread (i.e. xstream).